### ISSUE TITLE
Regression in 2.2.0 Cypher support

### ISSUE THREAD

**Issue by splilly** [neo4j/neo4j#4364]
Setup:

```
CREATE (a:root)-[:relates_to]->(b:leaf)-[:relates_to]->(c:leaf)
```

Query:

```
MATCH (a:root),(a)-[:relates_to]->(child:leaf),(a)-[*]->(decendants:leaf) RETURN *
```

This query returns results in 2.1.7 but in 2.2.0 I get back 0 rows.

Workaround:

```
MATCH (a:root),(a)-[:relates_to]->(child:leaf) WITH * MATCH (a)-[*]->(decendants:leaf) RETURN *
```

This query returns results in 2.1.7 and 2.2.0 and the results match the previous query for 2.1.7

**Comment by cleishm** [comment]
Thanks @splilly. I've verified this issue myself and have passed it along to our team.

**Comment by lutovich** [comment]
Hello @splilly,

Reported behaviour is an issue in 2.1.x that was fixed in 2.2.x.
Var-length pattern matching in 2.1.x does not respect the rule of relationship uniqueness in single `MATCH` clause.
Please check out this pull request that documents the problem in 2.1 branch: https://github.com/neo4j/neo4j/pull/4557.
Sorry for inconvenience.

**Comment by lutovich** [comment]
@splilly, this docs link mentions the relationship uniqueness rule: http://neo4j.com/docs/2.2.1/cypherdoc-uniqueness.html
